sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -171,6 +172,7 @@ private[sql] class SharedState(
 }
 
 object SharedState extends Logging {
```

Comment:
In practice, `SharedState` should be global per JVM. But what goes wrong if it isn't?

Reply:
If we want a new instance of `SparkSession`, we will always a `SharedState` too in all the existing APIs